### PR TITLE
add json dump

### PR DESCRIPTION
**Pull request by armintaenzertng**
This adds an option to dump the parsed spec information into a json file. This will enable anyone to easily create something from the specification without having to parse the markdown files first.

### CODE REVIEW

**Comment by fholger** [comment]
Currently, all output fields are written to the json as arrays, even those that should only have a singular value (e.g. `name`, `type`, ...). If possible, it would be great to fix that as it would make parsing and using the json output more convenient.

**Comment by armintaenzertng** [comment]
This is a general issue with the spec-parser, see #55.
@zvr, some clarification on this would be great! :)

**Review by maxhbr — APPROVED** [review]
It is an easy implementation, that will work for now. LGTM

**Review by zvr — APPROVED** [review]
I don't see the point, as I don't envision anyone else working with this input, but it could be useful for debugging -- and I'm sure others like JSON 🤣

**Comment by zvr** [comment]
We will be generating OWL and JSON-schema, which are standards,
Having a JSON dump and expect it to be actually used means we have to document it, maintain it, etc.

**Comment by maxhbr** [comment]
it is technical data and the code might count as documentation here.

And the OWL and JSON-Schema are both hard to parse.

**Comment by armintaenzertng** [comment]
We already have the first use case where @davaya implemented this functionality himself. In order not to split our efforts across multiple implementations that would eventually go out of sync, I referred him here. We might want to get this merged soon, then. :)

**Comment by davaya** [comment]
> Having a JSON dump and expect it to be actually used means we have to document it, maintain it, etc.

and

> Also, if someone wants to reason about the model data, one does not have to start to parse weird md files.

Exactly!  The JSON dump isn't an artifact to be maintained or even committed to a repo, JSON data is just a snapshot of the in-memory variables an application would have after parsing the "weird md files".  To be useful the JSON data **can't** be an artifact, it has to be live data generated directly from the model; the only artifact is the parser code.

(And the md files aren't weird - they're a brilliant way of representing the single source of truth from which everything else is generated, in a modular easy-to-maintain and easy to understand format.  They could be individual JSON or YAML files to save a little parsing effort, but the simple rigidity of the md bullet format forces designers to stay simple.  That's a feature, not a bug.)
